RANes/src/fds.cpp

```diff
@@ -690,34 +689,36 @@ int FDSLoad(const char *name, FCEUFILE *fp) {
 
 	FCEU_fseek(fp, 0, SEEK_SET);
 
-	fds_size = fp->size;
-	fds_ROM = (uint8*)FCEU_malloc(fds_size);
+	FDSSize = fp->size;
 
 	FreeFDSMemory();
 	if (!SubLoad(fp)) {
-		free(fds_ROM);
 		if(FDSBIOS)
 			free(FDSBIOS);
 		FDSBIOS = NULL;
 		return(0);
 	}
 
+	// Save original dump data to a separate buffer
+	FDSROM = (uint8*)FCEU_malloc(FDSSize);
```

Comment:
If `FDSSize` is not a strict multiple of 65500, you'll end up copying the wrong amount of data below. Recommend allocating `65500*TotalSides`

Reply:
True; done.